File: app/src/main/java/com/ytempest/wanandroid/interactor/configs/base/AbsConfig.java
package com.ytempest.wanandroid.interactor.configs.base;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * @author heqidu
 * @since 2020/8/15
 */
public abstract class AbsConfig {

    protected final PreferencesExtender mPref;

    public AbsConfig(Context context, @PreferencesName String name) {
        SharedPreferences preferences = context.getApplicationContext()
                .getSharedPreferences(name, Context.MODE_PRIVATE);
        mPref = createPreferencesExtender(preferences);
    }

    protected PreferencesExtender createPreferencesExtender(SharedPreferences preferences) {
        return new PreferencesExtender(preferences);
    }

}

File: app/src/main/java/com/ytempest/wanandroid/interactor/configs/base/AbsConfig.kt
package com.ytempest.wanandroid.interactor.configs.base

import android.content.Context
import android.content.SharedPreferences

/**
 * @author heqidu
 * @since 21-2-8
 */
abstract class AbsConfig(
        context: Context,
        @PreferencesName name: String, ) {

    protected val mPref: PreferencesExtender;

    init {
        val preferences = context.applicationContext.getSharedPreferences(name, Context.MODE_PRIVATE)
        mPref = createPreferencesExtender(preferences)
    }

    protected open fun createPreferencesExtender(preferences: SharedPreferences): PreferencesExtender {
        return PreferencesExtender(preferences)
    }

}